reflection/serverreflection.go

```diff
@@ -351,11 +311,12 @@ func (s *serverReflectionServer) fileDescEncodingContainingSymbol(name string, s
 	if fd == nil {
 		// Check if it's a type name that was not present in the
 		// transitive dependencies of the registered services.
-		if st, err := typeForName(name); err == nil {
-			fd, err = s.fileDescForType(st)
-			if err != nil {
-				return nil, err
-			}
+		desc, err := protoregistry.GlobalTypes.FindMessageByName(protoreflect.FullName(name))
+		if err != nil && err != protoregistry.NotFound {
+			panic(err)
```

Comment:
Return the error instead of `panic`

And why is `protoregistry.NotFound` not considered an error?